mmv1/products/dataproc/Batch.yaml

```
@@ -0,0 +1,504 @@
+            description: |
+              The Cloud KMS key to use for encryption.
+          - !ruby/object:Api::Type::String
+            name: 'idleTtl'
+            description: |
+              Applies to sessions only. The duration to keep the session alive while it's idling.
+          - !ruby/object:Api::Type::String
+            name: 'ttl'
+            description: |
+              The duration after which the workload will be terminated.
+          - !ruby/object:Api::Type::String
+            name: 'stagingBucket'
+            description: |
+              A Cloud Storage bucket used to stage workload dependencies, config files, and store
+              workload output and other ephemeral data, such as Spark history files.
+          - !ruby/object:Api::Type::String
+            name: 'networkUri'
+            description: |
+              Network configuration for workload execution.
+            exactly_one_of:
+              - network_uri
+              - subnetwork_uri
+          - !ruby/object:Api::Type::String
+            name: 'subnetworkUri'
+            description: |
+              Network configuration for workload execution.
+            exactly_one_of:
+              - network_uri
+              - subnetwork_uri
+      - !ruby/object:Api::Type::NestedObject
+        name: 'peripheralsConfig'
+        description: |
+          Peripherals configuration that workload has access to.
+        default_from_api: true
+        allow_empty_object: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'metastoreService'
+            description: |
+              Resource name of an existing Dataproc Metastore service.
+          - !ruby/object:Api::Type::NestedObject
+            name: 'sparkHistoryServerConfig'
+            description: |
+              The Spark History Server configuration for the workload.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: 'dataprocCluster'
+                description: |
+                  Resource name of an existing Dataproc Cluster to act as a Spark History Server for the workload.
+  - !ruby/object:Api::Type::String
+    name: 'operation'
+    description: |
+      The resource name of the operation associated with this batch.
+    output: true
+  - !ruby/object:Api::Type::Array
+    name: 'stateHistory'
+    description: |
+      Historical state information for the batch.
+    output: true
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::Enum
+          name: 'state'
+          description: |
+            The state of the batch at this point in history.
+          values:
+            - :STATE_UNSPECIFIED
+            - :PENDING
+            - :RUNNING
+            - :CANCELLING
+            - :CANCELLED
+            - :SUCCEEDED
+            - :FAILED
+        - !ruby/object:Api::Type::String
+          name: 'stateMessage'
```

I think this is not strictly necessary but would be nice for clarity.

```suggestion
          name: 'stateMessage'
          output: true
```